### ISSUE TITLE
Integration request: https://github.com/Birch-san/k-diffusion/tree/mps/k_diffusion

### ISSUE THREAD

**Issue by Gitterman69** [FahimF/sd-gui#11]
Can you integrate this diffusion? It is super optimised!

[BIRCH SAN - K DIFFUSION](https://github.com/Birch-san/k-diffusion/tree/mps/k_diffusion)

**Comment by Gitterman69** [comment]
Is there a quick way to integrate it myself? can't find the diffusion folders in the folder.... seems the structure is a bit different

**Comment by FahimF** [comment]
Yeah, Stable Diffusion implementations that use K Diffusion are a totally other beast, sorry 🙂 My implementation is based on Hugging Face Diffusers and so has to use the pipelines and schedulers that Hugging Face provides ... or write my own. This GUI is only about a week or two old and so I am at the point that I'm re-considering the architecture. But probably not till next weekend since I don't get to do a lot of work on this till the weekend.

If I do switch to the other approach then K Diffusion should be simple enough to add in.

**Comment by Gitterman69** [comment]
Nice to hear! The only gui alternative on Mac is currently diffusion bee but it is laughable.... no settings at all... BUT it is MPS optimised....

any tips how to get your ui running with other stable diffusion installations on my computer?

**Comment by FahimF** [comment]
Nothing that would work out of the box I'm afraid. Because the server expects certain parameters and the app will receive responses back from the server. So the parameters have to be named correctly in both cases. And each developer will simply use whatever names they wish. So without taking a look at each one and doing some work it's hard to make it work immediately would be my gut instinct ... without having taken a look at each server in question that is 🙂

I saw your other post about MPS optimizations. There are no optimizations that need to be applied as such. All of the MPS speed ups come from Pytorch. So as long as you have the latest Pytorch nightly (and my code is set to use MPS) then it should all just work.
